chainer/functions/pooling_2d.py

```diff
@@ -352,3 +355,100 @@ def average_pooling_2d(x, ksize, stride=None, pad=0, use_cudnn=True):
 
     """
     return AveragePooling2D(ksize, stride, pad, False, use_cudnn)(x)
+
+
+class SpatialPyramidPooling2D(function.Function):
+
+    """Spatial pyramid pooling over a set of 2d planes."""
+
+    def __init__(self, x_shape, pyramid_height, pooling_class, use_cudnn=True):
+        bottom_c, bottom_h, bottom_w = x_shape
+        assert bottom_h > 3 and bottom_w > 3, \
+            'input feature map size should be larger than 3'
+        self.pyramid_height = pyramid_height
+
+        # create pooling functions for different pyramid levels
+        split_inds = []
+        out_dim = 0
+        self.poolers = []
+        for pyramid_level in six.moves.range(pyramid_height):
+            num_bins = int(2 ** pyramid_level)
+
+            ksize_h = numpy.ceil(bottom_h / (float(num_bins)))
+            remainder_h = ksize_h * num_bins - bottom_h
+            pad_h = remainder_h // 2
+
+            ksize_w = numpy.ceil(bottom_w / (float(num_bins)))
+            remainder_w = ksize_w * num_bins - bottom_w
+            pad_w = remainder_w // 2
+
+            ksize = (int(ksize_h), int(ksize_w))
+            pad = (int(pad_h), int(pad_w))
+
```

Comment:
I think the following is better because we can reduce float calculation.
```
ksize_h = int(numpy.ceil(bottom_h / (float(num_bins)))) # add int()
remainder_h = ksize_h * num_bins - bottom_h
pad_h = remainder_h // 2

# (same for ***_w)

ksize = (ksize_h, ksize_w) # remove int()
pad = (pad_h, pad_w) # remove int()
```